docs/advanced/pool.rst

```diff
@@ -284,6 +284,45 @@ working connections, as soon as they are available.
 
 Faster than you can say poll. Or pool.
 
+Row factories
+-------------
+
+Pool's connections may be configured to use a :ref:`row factory
+<row-factories>`. One approach is to use the `!configure` constructor as
+follows:
+::
+
+    def set_row_factory(conn: psycopg.Connection[Any]) -> None:
+        conn.row_factory = psycopg.rows.class_row(MyRow)
+
+    pool = ConnectionPool[MyRow](configure=set_row_factory)
+    with pool.connection() as conn:
+        row = conn.execute(...).fetchone()
+        # row has type MyRow (or None)
+        # conn has type Connection[MyRow]
+
+Here it is needed to set ``ConnectionPool``'s type parameter to ``MyRow`` as
+the `!configure` callback is not enough for type inference.
+
+The same can be achieved by passing ``kwargs={"row_factory":
+psycopg.rows.class_row(MyRow)}`` to ``ConnectionPool``, but the type parameter
+for ``ConnectionPool`` remains needed.
+
+Alternatively, the `!connection_class` constructor may be used:
+::
+
+    class MyConnection(psycopg.Connection[MyRow]):
+        def __init__(self, *args):
+            super().__init__(*args)
+            self.row_factory = psycopg.rows.class_row(MyRow)
```

Comment:
Somehow unrelated: writing this made me realize that this not very convenient; I'd have appreciated if the following worked:
```python
class MyConnection(psycopg.Connection[MyRow]):
    row_factory = psycopg.rows.class_row(MyRow)
```
and, at a first glance, I think it can be made working with a bit of work.
Thoughts?

Reply:
If I understand correctly, this could be a mistake, of the type I've already made, for instance for `Dumper.oid`, which is a class attribute, but can be overridden by instances. It is valid python, but Mypy doesn't like it very much. It cannot be annotated as a ClassVar and mypyc is confused as of whether it's a class or instance attribute. So I would like to refactor it away in psycopg 3.2 (if I can do it without losing backward compatibility).